### ISSUE TITLE
Give us an option to disable drag-n-drop in editor box

### ISSUE THREAD

**Issue by yumeyao** [mattgodbolt/gcc-explorer#145]
or at least, don't overwrite clipboard when drag-n-drop happens.

This extremely hurts user experience fluency.
consider you have some other code piece in other window/editor/etc. and you want to use this code piece to replace some code piece in the editor box.
1. select other code piece
2. Ctrl-C
3. select code piece to be replaced in the editor box
4. Ctrl-V -----> doesn't work because the drag-n-drop action(very easily to happen when you want to select something) overwrites clipboard with what you just select.

**Comment by mattgodbolt** [comment]
Hi: I don't see this. If I ctrl-C some code, then select some other code and ctrl-V over it, the pasted text replaces the selected text.  Can you clarify please? What browser/OS? 

Perhaps I misunderstand: drag and drop is when you select text, then drag that selected text around. It doesn't involve the clipboard at all?

**Comment by yumeyao** [comment]
tested on Chrome/IE11 under Windows 7.

yes, drag-n-drop is what exactly you suggested. the point is, sometimes when you want to select some text, it becomes drag-n-drop (I think the reason for this is you want to select something while there is some pending compile in progress, when the compile finishes, some scripts updates the input box that interrupts the selection behavior.

so it would be
1. select other code piece
2. Ctrl-C
3. **drag-n-drop something** (you want to just select code piece to be replaced, but maybe as the reason above, ocassionally it becomes drag-n-drop)
4. Ctrl-V -----> doesn't work because the drag-n-drop action(very easily to happen when you want to select something) overwrites clipboard with what you just select.

**Comment by mattgodbolt** [comment]
Thanks for clarifying the issue! I'll take a look, thanks!

**Comment by mattgodbolt** [comment]
I just took a look and for the life of me I can't get any interaction to happen between the clipboard and any drag and drop. I'm on Linux, so maybe that makes a difference. With no compiles happening, I select text then copy it.  Then I select different text, and drag it elsewhere in the document.  Finally, I paste, and the original text I copied is pasted.  I'll keep trying but seems whatever's going on I can repro easily.

I may just disable drag and drop of code: I doubt anyone will miss it and it may fix your issue!
